clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1757,6 +1771,20 @@ private void subscribeInternal(Pattern pattern, Optional<ConsumerRebalanceListen
         }
     }
 
+    private void subscribeInternal(SubscriptionPattern pattern, Optional<ConsumerRebalanceListener> listener) {
+        acquireAndEnsureOpen();
+        try {
+            maybeThrowInvalidGroupIdException();
+            if (pattern == null) {
+                throw new IllegalArgumentException("Topic pattern to subscribe to cannot be null");
+            }
+            log.info("Subscribed to pattern: '{}'", pattern.pattern());
+            subscriptions.subscribe(pattern, listener);
```

Comment:
We've made a lot of effort to move any mutation of the `SubscriptionState` object to the background thread. Is there any reason we should not follow the same pattern (pun intended) here? If there's a good reason not to, we definitely should add some documentary comments to that effect.